go/vt/vttablet/tabletserver/connpool/dbconn_test.go

```diff
@@ -320,66 +320,153 @@ func TestDBKillWithContext(t *testing.T) {
 		time.Sleep(200 * time.Millisecond)
 	})
 
-	ctx, cancel := context.WithTimeout(context.Background(), 100*time.Millisecond)
-	defer cancel()
+	// set a lower timeout value
+	dbConn.killTimeout = 100 * time.Millisecond
 
-	// KillWithContext should return context.DeadlineExceeded
-	err = dbConn.KillWithContext(ctx, "test kill", 0)
+	// Kill should return context.DeadlineExceeded
+	err = dbConn.Kill("test kill", 0)
 	require.ErrorIs(t, err, context.DeadlineExceeded)
 }
 
-func TestDBKillWithContextDoneContext(t *testing.T) {
-	db := fakesqldb.New(t)
-	defer db.Close()
-	connPool := newPool()
-	params := dbconfigs.New(db.ConnParams())
-	connPool.Open(params, params, params)
-	defer connPool.Close()
-	dbConn, err := newPooledConn(context.Background(), connPool, params)
-	if dbConn != nil {
-		defer dbConn.Close()
+// TestDBConnCtxError tests that an Exec returns with appropriate error code.
+// Also, verifies that does it wait for the query to finish before returning.
+func TestDBConnCtxError(t *testing.T) {
+	exec := func(ctx context.Context, query string, dbconn *Conn) error {
+		_, err := dbconn.Exec(ctx, query, 1, false)
+		return err
 	}
-	require.NoError(t, err)
 
-	query := fmt.Sprintf("kill %d", dbConn.ID())
-	db.AddRejectedQuery(query, errors.New("rejected"))
+	execOnce := func(ctx context.Context, query string, dbconn *Conn) error {
+		_, err := dbconn.ExecOnce(ctx, query, 1, false)
+		return err
+	}
+
+	t.Run("context cancel - non-tx exec", func(t *testing.T) {
+		ctx, cancel := context.WithCancel(context.Background())
+		go func() {
+			time.Sleep(10 * time.Millisecond)
+			cancel()
+		}()
+		testContextError(t, ctx, exec,
+			"(errno 1317) (sqlstate 70100): Query execution was interrupted",
+			150*time.Millisecond)
```

Comment:
I think that this test is likely to be flaky in the CI, given these very small time windows. I'd recommend extending all of them at least a bit.

Reply:
I have run these tests multiple times and have not observed any flakiness. I will increase them if they come out as flaky.